### PR TITLE
update workflow configuration for testing websites in pull requests

### PR DESCRIPTION
**Pull request by awentzel**
# Pull Request

## 📖 Description
FAST websites have stopped pushing builds to Static Web Apps due to the limitation of 10 builds per Static Web App.  To work around this limitation, this change will only build the website when `sites/**/[site-name]` source code has changed. 
* Component Explorer
* Documentation
* Color Explorer

### 🎫 Issues
* Static Web Apps in PRs used to conveniently and quickly test websites have stopped working.
* Static Web App limits are being reached
* Static Web App deployment should be deleted upon PR closure
* Only deploy if the site files have changed
* Both explorers were missing destroy pr's, adding back in.

## 👩‍💻 Reviewer Notes
Testing procedures
1. Test pushing a change outside this package (expect ignore)
2. Test pushing a change inside this package (expect trigger)
3. Test pushing a workflow file change (expect trigger)
4. Test pushing a different workflow file (expect ignore)
5. Test with 10 deployment limit (expect failure)
6. Test with < 10 deployment limit (expect pass)
7. Test Close PR will delete the deployment. So if deploying item A on PR open, then item A on PR close should be deleted, thus we never exceed limitation unless too many PRs exist on any particular site.
* I may be required to manually delete deployments on Azure to get under the 10 limit in order to test.

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

## ⏭ Next Steps
Upon completion this same pattern will be implemented across websites deployed from FAST.

### CODE REVIEW

File: .github/workflows/cd-deploy-www-staging.yml [hunk 1]
```diff
@@ -3,10 +3,17 @@ on:
...
       - master
+    paths:
+      - 'sites/website/**'
+      - '.github/workflows/cd-deploy-www-staging.yml'
   pull_request:
     types: [opened, synchronize, reopened, closed]
     branches:
       - master
+    paths:
+      - 'sites/website/**'
```
Comment (janechu) [inline .github/workflows/cd-deploy-www-staging.yml:14]: Should we include the frontpage?
```suggestion
      - 'sites/fast-website/**'
      - 'sites/website/**'
```

**Review by janechu — APPROVED** [review]

File: .github/workflows/cd-deploy-www-staging.yml [hunk 2]
```diff
@@ -3,10 +3,19 @@ on:
   push:
     branches:
       - master
+    paths:
+      - 'sites/website/**'
```
Comment (EisenbergEffect) [inline .github/workflows/cd-deploy-www-staging.yml:7]: Is it possible to also detect when any markdown file or api-extractor.json file changes? Any changes in these technically also would effect the content on the documentation site?

Comment (awentzel) [reply .github/workflows/cd-deploy-www-staging.yml:8]: Updated for www to include any markdown changes using `**.md` as referenced from https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#patterns-to-match-file-paths cheat sheet.

**Comment by nicholasrice** [comment]
Does this mean our documentation site won't update when API extractor info from fast-element and fast-foundation are changed?

**Comment by nicholasrice** [comment]
I just want to confirm when you say "any change" that you're not meaning any change _tracked by git_, is that accurate? I'm asking because there are a lot of .md files feeding the documentation site that live outside the sites directory (they live in fast-element, fast-foundation), and we have build scripts to bring them over into the sites directory; those built files are not tracked by git.

**Comment by awentzel** [comment]
A meeting has been scheduled among a few of the contributors to discuss best path forward.

**Comment by EisenbergEffect** [comment]
I think we'll want to trigger the docs build whenever there is a change in the api.json files as well. If we get that, the markdown, and the content of the sites folder, I think that's all the needed paths. @williamw2 may know if there are any other files needed, based on his recent component API generation work.

**Comment by williamw2** [comment]
Yes. Those would be the README.md files in the fast-foundation/src/<ComponentName> folders. These files are tracked by Git and are pulled over to the website during the build. These files become the Components pages on the website.

**Review by nicholasrice — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]

**Pull request merged**